text/000-failures.md

```diff
@@ -26,47 +26,89 @@ A failure can be repaired. The realism level could be a configuration item. When
+
+The consumer will read _SV<sub>misuse</sub>_ and overwrite it with _0_ after reading. The publisher itself will then either remove the lock written to _SV<sub>lock</sub>_ or write another _F<sub>i</sub>_ into _SV<sub>misuse</sub>_. The responsibility of removing the lock remains with the publisher, as giving this responsibility to the consumer could lead to a situation where the publisher reads _SV<sub>lock</sub>_ and _SV<sub>misuse</sub>_, with the following result: _(ID, 0)_, and then incorrectly assumes it can write the next _F<sub>i</sub>_, while in fact _SV<sub>lock</sub>_ has become _0_ right after the read.
+
+### Recovering from failures
+
+A case not yet covered is that of failure recovery itself, achieved by performing various actions on the piece(s) of equipment. For example, turning an engine generator OFF and then ON again might recover its function. Failure recovery follows the same communication pattern as described in the [stress/misuse trigger](#stressmisuse-trigger) section above, with the additional of a _SV<sub>recovery</sub>_ SimVar.
+
+## Failure ranges
+
+From a software point of view, it does not matter which numeric identifier _F<sub>i</sub>_ is used for which failure. For ease of understanding however, using ranges per area is helpful (for example ELEC, HYD). For example:
+
+_ELEC = F<sub>1</sub>,⋯,F<sub>999</sub>_
+
+_HYD = F<sub>1000</sub>,⋯, F<sub>1999</sub>_
+
+Should an area ever exceed the range allotted, another range can be given to it.
+
+## Systems.wasm
+
+The Rust systems code has a lot of functionality which is generic. A failure code on the A320 is unlikely to be the same as a failure code on the A380. Thus, care should be taken to translate the failure code within the `a320_systems_wasm` crate to a failure message that is generic.
+
+Within systems.wasm, a piece of equipment is a (part of a) `SimulationElement`.
+
+On first thought, one could consider using the `SimulationElement.read` and `SimulationElement.write` functions for reading and writing failure information within each `SimulationElement`. However, this leads to various small problems:
+
+- Unlike simulation variables changes, failures happen rarely, and thus continuously reading failure information is inefficient.
+- Failures due to stress/misuse should not be written towards the sim every frame. They should only be written once. By including this in the `SimulationElement.write`, we would give the responsibility for not repeating writes to each individual element.
+
+As such, a more elaborate approach is probably warranted. A potentially better solution is introducing a `Failure` type of zero or more instances are stored within each element.
+
+The `Failure` type would contain the following capabilities:
+
+- A function to query whether the failure is active.
+- A function through which the element can indicate the failure became active (due to stress/misuse).
```

Comment:
We should look into communicating this to a simvar so we can display that in the EFB.

Reply:
That's the plan.